### PR TITLE
doc: remove "prior to Homestead" descriptions

### PR DESCRIPTION
**Pull request by pirapira**
In the documentation there were two sidenotes about "prior to Homestead" era.   I think these notes are no longer necessary.

The other change is just a clarification about when the function overload resolution is performed.

### CODE REVIEW

**Comment by axic** [comment]
If we get #1117 implemented, this may become useful?

**Comment by pirapira** [comment]
After #1117, in my opinion, EVM version differences should be explained in a special section about the differences.  The other parts of the documentation can always talk about the "current" one.

**Comment by chriseth** [comment]
I think we should still leave that in, people might ask about old rules and opcodes.

File: docs/contracts.rst [hunk 1]
```diff
@@ -976,8 +975,7 @@ are all compiled as calls (``DELEGATECALL``) to an external
 contract/library. If you use libraries, take care that an
 actual external function call is performed.
 ``msg.sender``, ``msg.value`` and ``this`` will retain their values
-in this call, though (prior to Homestead, ``msg.sender`` and
-``msg.value`` changed, though).
```
Comment (axic) [inline docs/contracts.rst:980]: Actually I think we should extend this with an explanation: "because of the use of `CALLCODE` and not `DELEGATECALL`"

File: docs/contracts.rst [hunk 2]
```diff
@@ -1079,7 +1077,7 @@ In both situations, all functions, even those where the
 type of the first parameter does not match the type of
 the object, are attached. The type is checked at the
 point the function is called and function overload
-resolution is performed.
+resolution is performed at the compilation time.
```
Comment (axic) [inline docs/contracts.rst]: Is this related to this PR?

Comment (pirapira) [reply docs/contracts.rst]: No, but I didn't even split the commits.  hmm.

**Comment by pirapira** [comment]
I thought these phrases are leftovers from the era when people remembered things before Homestead, but from the comments here, I guess that's not the case.

My main concern was about undefined names like "Homestead" appearing in the middle of a text.
